community/cypher/frontend-3.0/src/main/scala/org/neo4j/cypher/internal/frontend/v3_0/ast/Clause.scala

```diff
@@ -203,17 +203,25 @@ case class Match(optional: Boolean, pattern: Pattern, hints: Seq[UsingHint], whe
 }
 
 case class Merge(pattern: Pattern, actions: Seq[MergeAction])(val position: InputPosition) extends UpdateClause {
-  def name = "MERGE"
+  override def name = "MERGE"
 
-  def semanticCheck =
+  override def semanticCheck =
     pattern.semanticCheck(Pattern.SemanticContext.Merge) chain
-      actions.semanticCheck
+      actions.semanticCheck chain checkRelTypes
+
+  // Copied code from CREATE below
```

Comment:
Would it not be better to use the same code in both places instead of adding a comment saying that it is copied?